### ISSUE TITLE
TypeScript: Prettier gets confused about "this" when used as a JSX element

### ISSUE THREAD

**Issue by prencher** [prettier/prettier#1877]
`this` incorrectly gets rewritten as `undefined` in the following code:
```tsx
export class Loadable extends React.Component<Props, State> {
  public render() {
    return <this.state.Component {...this.props.componentProps} />;
  }
}

```

Output:
```tsx
export class Loadable extends React.Component<Props, State> {
  public render() {
    return <undefined.state.Component {...this.props.componentProps} />;
  }
}
```

**Comment by vjeux** [comment]
Awesome, please keep those coming! I'll release a 1.4.1 with those fixes.

**Comment by vjeux** [comment]
This is an issue in the underlying project, but we should be able to workaround.
https://github.com/eslint/typescript-eslint-parser/issues/307

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!
